### PR TITLE
Add assets and java resources data collection

### PR DESCRIPTION
**Pull request by Temennigru**
Also remove hard coded allowed extensions

Test: Existing

### CODE REVIEW

File: generator/src/main/kotlin/com/android/gradle/replicator/generator/project/GradleProjectGenerator.kt [hunk 1]
```diff
@@ -241,7 +250,7 @@ class GradleProjectGenerator(
         val generationPropertyFile = folder.join("generation.properties")
 
         val loader = Thread.currentThread().contextClassLoader
-        loader.getResourceAsStream("project/resourceGeneratorConstants.properties")!!.copyTo(FileOutputStream(generationPropertyFile))
+        loader.getResourceAsStream("project/resource-generator-constants.properties")!!.copyTo(FileOutputStream(generationPropertyFile))
```
Comment (jdochez) [inline generator/src/main/kotlin/com/android/gradle/replicator/generator/project/GradleProjectGenerator.kt:253]: using getResourceAsStream will leak the jar file and class loader if not closed properly

File: model/src/main/kotlin/com/android/gradle/replicator/model/internal/filedata/FileData.kt [hunk 2]
```diff
@@ -167,7 +157,7 @@ class AndroidResourcePropertiesAdapter(private val resourceType: String? = null)
                 qualifiers = qualifiers!!,
```
Comment (jdochez) [inline model/src/main/kotlin/com/android/gradle/replicator/model/internal/filedata/FileData.kt:157]: something to do in a subsequent CL. I am not a big fan of using nullable and then the !! operator for what is expected to be present. Instead you should use a Builder class that has a bunch of setXXX() like for instance setResourceType(resourceType: String), and setFileData(fileData: List<Long>) and only check that all mandatory values are present when build() is called.

**Review by jdochez — CHANGES_REQUESTED** [review]

File: code/codegen/src/main/kotlin/com/android/gradle/replicator/resgen/Main.kt [hunk 3]
```diff
@@ -158,6 +179,14 @@ class Main {
         return count
     }
 
+    private fun countFilesWithSizeMetadata(files: FilesWithSizeMap): Int {
```
[the remark this answers is not in the document]
Comment (Temennigru) [reply code/codegen/src/main/kotlin/com/android/gradle/replicator/resgen/Main.kt:188]: This does count the files.

FileSizeMap is a map of extensions to a list of file sizes like this:
`{
  "xml": [2, 4, 6, 8],
  "png": [10, 12, 14, 16]
}`

so my loop goes through the extensions and adds up the *number* of files in each one (extension.value is the list, and extension.value.size is the number of elements in the list).

File: model/src/main/kotlin/com/android/gradle/replicator/model/internal/filedata/FileData.kt (hunk 2 above)
Comment (Temennigru) [reply model/src/main/kotlin/com/android/gradle/replicator/model/internal/filedata/FileData.kt:157]: This is simple enough that I can add it to this CL. Let me know what you think of my implementation

**Review by jdochez — APPROVED** [review]

**Pull request merged**